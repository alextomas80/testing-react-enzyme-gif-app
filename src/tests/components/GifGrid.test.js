import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el <GifGrid />', () => {
	const category = 'Bart Simpson';

	test('debe de mostrarse correctamente', () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true,
		});

		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	});

	test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
		const gifs = [
			{
				id: 'ABC',
				url: 'https://xxx/cualquier/cosa.jpg',
				title: 'Una cosa',
			},
			{
				id: '123',
				url: 'https://xxx/cualquier/cosa.jpg',
				title: 'Otra cosa',
			},
		];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});

		const wrapper = shallow(<GifGrid category={category} />);

		expect(wrapper.find('p').exists()).toBe(false);
		expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
	});
});
