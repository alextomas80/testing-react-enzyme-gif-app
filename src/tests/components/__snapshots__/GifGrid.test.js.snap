// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="category-name  animate__animated animate__fadeIn"
  >
    Bart Simpson
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando gifs, un momento...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
